src/main/python/hydra/lib/hydrabase.py

```diff
@@ -666,6 +671,66 @@ def get_mesos_slave_ips_attr(self, attr_type, attr_value):
         """
         return self.__mesos.get_slave_ips_from_attribute(attr_type, attr_value)
 
+    def get_all_mesos_slave_attr(self):
```

Comment:
Is it necessary to put dictionary datastructure info to list. Can't we just do something like below

for idx, info in self.mesos_cluster.items():
  self.get_mesos_slave_ips_attr(info["cat"], info["match"])

Lesser number of functions (data structures) will save time needed to understand the code.